app/Http/Controllers/Front/PageController.php

```diff
@@ -719,28 +739,29 @@ public function getmonthPriceDescription(int $productid)
         try {
```

Comment:
public function getmonthPriceDescription(int $productid)
{
    try {
        $product = Product::find($productid);

        if (!$product || $product->add_to_contact == 1) {
            return '';
        }

        $priceDescription = '';

        $plans = Plan::where('product', $productid)
            ->where('days', '>=', 30) // Consider plans with 30 or more days
            ->with('planPrice')
            ->get();

        foreach ($plans as $plan) {
            $description = $plan->planPrice->first();

            if ($description && $description->price_description == 'Free') {
                $priceDescription = 'free';
            } else {
                // Determine price description based on plan details
                $priceDescription = $description ?
                    ($description->no_of_agents ?
                        'per month for <strong>' . $description->no_of_agents . ' agent</strong>'
                        : 'per month')
                    : '';
            }

            if ($priceDescription) {
                break; // Stop loop once a valid description is found
            }
        }

        return $priceDescription;
    } catch (\Exception $ex) {
        app('log')->error($ex->getMessage());
        return redirect()->back()->with('fails', $ex->getMessage());
    }
}

Try this I also want to know why this function is used. It it reasonable to check description  = free

Reply:
this function is for fetching price description for monthly products.And it is reasonable to check if the description equals "Free" as part of determining the price description.